File: src/main/scala/org/xbery/artbeams/common/assets/domain/Asset.scala
package org.xbery.artbeams.common.assets.domain

import java.time.Instant

/**
 * Common superclass of all content entities.
  *
  * @author Radek Beran
 */
abstract class Asset extends Serializable {

  def common: AssetAttributes

  // Convenience methods accessing common attributes
  def id: String = common.id
  def created: Instant = common.created
  def createdBy: String = common.createdBy
  def modified: Instant = common.modified
  def modifiedBy: String = common.modifiedBy
}

File: src/main/kotlin/org/xbery/artbeams/common/assets/domain/Asset.kt
package org.xbery.artbeams.common.assets.domain

import org.xbery.artbeams.common.repository.IdentifiedEntity
import java.io.Serializable
import java.time.Instant

/**
 * Common superclass of all content entities.
 *
 * @author Radek Beran
 */
abstract class Asset : Serializable, IdentifiedEntity {
    abstract val common: AssetAttributes
    override val id: String get() = common.id
    val created: Instant get() = common.created
    val createdBy: String get() = common.createdBy
    val modified: Instant get() = common.modified
    val modifiedBy: String get() = common.modifiedBy
}
